src/GNDStk/convert/src/HDF5.hpp

```diff
@@ -0,0 +1,188 @@
+
+// -----------------------------------------------------------------------------
+// convert(*,HDF5)
+// That is, convert to HDF5 objects
+// -----------------------------------------------------------------------------
+
+// -----------------------------------------------------------------------------
+// Node ==> HDF5
+// -----------------------------------------------------------------------------
+
+inline bool convert(const Node &node, HDF5 &h)
+{
+   // clear
+   h.clear();
+
+   // for the HDF5
+   h.filename = h.temporaryName();
+
+   static const std::string context = "convert(Node,HDF5)";
+   try {
+
+      // Open temporary HDF5 file. This *should* work, but we check anyway.
+      std::ofstream ofs(h.filename, std::ios::binary);
+      if (!ofs) {
+         log::error("Could not open temporary HDF5 file \"{}\"", h.filename);
+         throw std::exception{};
+      }
+      ofs.close();
```

Comment:
I didn't actually check to see precisely what the HDF5 library does. Either way, I'd have my own "try to open it," so that the very first error message that's printed, if the open fails, is GNDStk's. This way, the look and feel of the HDF5 interface is the same as the look and feel of our XML and JSON interfaces. Generally speaking, GNDStk's XML, JSON, and HDF5 classes are designed to serve as lightweight wrappers around the classes in pugixml, nlohmann json, and highfive, and we do this precisely so that all the behavior can be given the same look and feel. The way I deal with diagnostics is just one aspect of this goal.